api/models/release.go

```diff
@@ -671,3 +668,93 @@ func waitForTemplate(bucket string, id string) error {
 	fmt.Printf("ns=kernel at=waitForTemplate.tick error=unknown bucket=%q release=%q\n", bucket, id)
 	return fmt.Errorf("unknown error")
 }
+
+// monitorReleasePromotion observes a release's rollout by verifying the app's CF stack status
+// Sends a notifcation on success or failure. This function blocks.
+// TODO: this should be in provider.ReleasePromote()
+func waitForPromotion(r *Release) {
```

Comment:
Can we do this with CloudFormation events?

We could create an SNS topic on the Rack that either calls back to a Rack HTTP endpoint (see webhooks) or have the Rack subscribe to an SQS queue that is subscribed to that SNS topic.

This should give us real "events" for CF updates (we could probably even use this to add things to the logs if we want to) without the polling which will fail if a rack container is restarted while waiting.